src/midi/midi_fluidsynth.cpp

```diff
@@ -44,6 +43,224 @@
+                                
+int  (*fluid_synth_write_float)(fluid_synth_t *synth, 
+                                int len,
+                                void *lout, 
+                                int loff, 
+                                int lincr,
+                                void *rout, 
+                                int roff, 
+                                int rincr) = nullptr;
+
+// clang-format on
+
+} // namespace fsynth
+
+/**
+ * A filthy macro to make resolving library symbols a little less verbose, 
+ * while still including robust error checking.
+ */
+#define DB_GET_FSYNTH_SYM(sym) sym = (decltype(sym))dynlib_get_symbol(fsynth_lib, #sym); \
+	if (!sym) { dynlib_close(fsynth_lib); return DynLibResult::ResolveSymErr; }
+
+/**
+ * Load the FluidSynth library and resolve all required symbols.
+ * 
+ * If the library is already loaded, does nothing.
+ * 
+ * IMPORTANT: If adding a new symbol above, remember to resolve 
+ * the symbol in this function, otherwise Dosbox is likely to segfault.
+ */
+static DynLibResult load_fsynth_dynlib()
+{
+	using namespace fsynth;
+	if (!fsynth_lib) {
+		fsynth_lib = dynlib_open(fsynth_dynlib_file);
+		if (!fsynth_lib) {
+			return DynLibResult::LibOpenErr;
+		}
+		DB_GET_FSYNTH_SYM(fluid_version);
+        DB_GET_FSYNTH_SYM(fluid_set_log_function);
+
+		DB_GET_FSYNTH_SYM(new_fluid_settings);
+		DB_GET_FSYNTH_SYM(new_fluid_synth);
+		
+		DB_GET_FSYNTH_SYM(delete_fluid_settings);
+		DB_GET_FSYNTH_SYM(delete_fluid_synth);
+		DB_GET_FSYNTH_SYM(fluid_settings_setnum);
+
+		DB_GET_FSYNTH_SYM(fluid_synth_chorus_on);
+		DB_GET_FSYNTH_SYM(fluid_synth_set_chorus_group_nr);
+		DB_GET_FSYNTH_SYM(fluid_synth_set_chorus_group_level);
+		DB_GET_FSYNTH_SYM(fluid_synth_set_chorus_group_speed);
+		DB_GET_FSYNTH_SYM(fluid_synth_set_chorus_group_depth);
+		DB_GET_FSYNTH_SYM(fluid_synth_set_chorus_group_type);
+
+		DB_GET_FSYNTH_SYM(fluid_synth_reverb_on);
+		DB_GET_FSYNTH_SYM(fluid_synth_set_reverb_group_roomsize);
+		DB_GET_FSYNTH_SYM(fluid_synth_set_reverb_group_damp);
+		DB_GET_FSYNTH_SYM(fluid_synth_set_reverb_group_width);
+		DB_GET_FSYNTH_SYM(fluid_synth_set_reverb_group_level);
+
+		DB_GET_FSYNTH_SYM(fluid_synth_sfcount);
+		DB_GET_FSYNTH_SYM(fluid_synth_sfload);
+		DB_GET_FSYNTH_SYM(fluid_synth_set_gain);
+		DB_GET_FSYNTH_SYM(fluid_synth_set_interp_method);
+		DB_GET_FSYNTH_SYM(fluid_synth_noteoff);
+		DB_GET_FSYNTH_SYM(fluid_synth_noteon);
+		DB_GET_FSYNTH_SYM(fluid_synth_key_pressure);
+		DB_GET_FSYNTH_SYM(fluid_synth_cc);
+		DB_GET_FSYNTH_SYM(fluid_synth_program_change);
+		DB_GET_FSYNTH_SYM(fluid_synth_channel_pressure);
+		DB_GET_FSYNTH_SYM(fluid_synth_pitch_bend);
+		DB_GET_FSYNTH_SYM(fluid_synth_sysex);
+		DB_GET_FSYNTH_SYM(fluid_synth_write_float);
+
+        /* Keep ERR and PANIC logging only */
+        for (auto level : {FLUID_DBG, FLUID_INFO, FLUID_WARN}) {
+            fluid_set_log_function(level, nullptr, nullptr);
+        }
```

Comment:
Indentation

Reply:
What's wrong with this?

Reply:
Seemed like the indentation was off, but probably I was confused. Probably I did not see the full context, just 2-3 preceding liens.